### PR TITLE
C build

### PR DESCRIPTION
**Pull request by DenverM80**
Modify ds3.c to use proper Request template based on request and / or response payload
Add hasRequestPayload and requestPayload properties to Request model

Implement HeaderConverter to sort order of structs defined in ds3.h

Namespace Enum values by prefixing EnumType; strip off this prefix when matching XML or printing enumToStr

### CODE REVIEW

File: ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/CCodeGenerator.java [hunk 1]
```diff
@@ -84,8 +81,8 @@ public void generateHeader(
             final ImmutableList<Struct> allStructs,
             final ImmutableList<Request> allRequests) throws IOException, ParseException {
         final Path path = Paths.get("src/ds3.h");
-        final Header header = new Header(allEnums,allStructs,allRequests);
-        processTemplate(header, "ds3_h.ftl", fileUtils.getOutputFile(path));
+        final Header header = HeaderConverter.toHeader(allEnums,allStructs,allRequests);
```
Comment (RachelTucker) [inline ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/CCodeGenerator.java]: missing space after comma

File: ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/converters/HeaderConverter.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import com.spectralogic.ds3autogen.c.models.Header;
+import com.spectralogic.ds3autogen.c.models.Request;
+import com.spectralogic.ds3autogen.c.models.Struct;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.text.ParseException;
+
+public class HeaderConverter {
+    private static final Logger LOG = LoggerFactory.getLogger(HeaderConverter.class);
```
Comment (RachelTucker) [inline ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/converters/HeaderConverter.java]: LOG is not used

**Comment by DenverM80** [comment]
This change set contains https://github.com/SpectraLogic/ds3_autogen/pull/199, and this diff will get a bit simpler if that's merged first

**Comment by DenverM80** [comment]
updated after merging from master

**Pull request merged**